addons/im_livechat/static/src/legacy/public_livechat.js

```diff
@@ -2927,6 +2927,7 @@ var ThreadWidget = Widget.extend({
         })));
 
         options = _.extend({}, modeOptions, options, {
+            messagesSeparatorPosition: false,
```

Comment:
It seems to be used in old qweb template only when specific 'top' value or it maps to a message id (integer).
How setting `false` instead of `undefined` fixed this issue?

Reply:
`message.getID()` return `undefined` when there is no message yet I think, I don't remember exactly

Reply:
Ok, I think it's best to investigate further why `message.getID()` returns `undefined`, because this shouldn't happen. Could be related to the "How may I help you?" message, but from a glance on code it has string `'_welcome'` as id so shouldn't be an issue I think...? Message ids have been considered as integers only, so wouldn't be surprised it's buggy.

Reply:
![image](https://user-images.githubusercontent.com/48204996/88914885-9ca6e500-d263-11ea-875a-69edec2fe009.png)

Don't know why but when writing something in the composer, a message is already created, with id undefined

Reply:
Ok problem is that livechat is handling notification whatever happen, so it handles typing notif like it was a message creation

Reply:
> Ok problem is that livechat is handling notification whatever happen, so it handles typing notif like it was a message creation

Seems to be the cause from task-2299331

Reply:
Indeed, also check task-2282271

Reply:
> Indeed, also check task-2282271

Don't think this is related.

Reply:
_welcome is created before, here it's created due to the typing notification, he's creating an empty message

Reply:
That's weird, could be a consequence of changes we have made in python, since we haven't change code in public livechat. Fix would consist of not creating this empty message from typing notification.

Reply:
I don't think so, I think the old notification handler was preventing the notification to go to livechat in some way because the code seems to assume that livechat can only get "history_command' notification or a normal messsage.

```js
_handleNotification: function (notification) {
    if (this._livechat && (notification[0] === this._livechat.getUUID())) {
        if (notification[1]._type === 'history_command') { // history request
            var cookie = utils.get_cookie(LIVECHAT_COOKIE_HISTORY);
            var history = cookie ? JSON.parse(cookie) : [];
            session.rpc('/im_livechat/history', {
                pid: this._livechat.getOperatorPID()[0],
                channel_uuid: this._livechat.getUUID(),
                page_history: history,
            });
        } else { // normal message
            // If message from notif is already in chatter messages, stop handling
            if (this._messages.some(message => message.getID() === notification[1].id)) {
                return;
            }
            this._addMessage(notification[1]);
            this._renderMessages();
            if (this._chatWindow.isFolded() || !this._chatWindow.isAtBottom()) {
                this._livechat.incrementUnreadCounter();
            }
        }
    }
},
```

Reply:
Here, in the "// normal message" case, `notification[1]` is in fact a typing notification and not a message like the code suppose it is

Reply:
> > Ok problem is that livechat is handling notification whatever happen, so it handles typing notif like it was a message creation
> 
> Seems to be the cause from task-2299331

> Indeed, also check task-2282271

I'm not sure but the problem in that tasks are notifications sent by livechat (or a problem in handling them in new code), here the problem is the handling of notification in the old code of livechat.
(That being said, what I changed fixed the problem but not the problems in those tasks you guys mentioned)

Reply:
(Just read what you said on task, don't bother to repeat yourself @alexkuhn ;p )

Reply:
in new composer written with OWL we have : 
```js
this.thread.unregisterCurrentPartnerIsTyping({ immediateNotify: true });
```

which is not done in livechat public code.
So basically, we need to call the RPC before calling chat_post in livechat.

Reply:
Made this in another PR as not fully linked to the task fixed by this one